Add examples for Re.split function

mooreryan opened the pull request:
Adds a few examples of using zero-length patterns as inputs to Re.split.  Examples are taken from the test suite.  Related to #119 and #120.

I know it is an old issues (this one: https://github.com/ocaml/ocaml-re/issues/119), but I also tried to use the `eol` as input for `split` in the past.  So, I took a couple of the test cases and added them to the examples for `Re.split`.

mooreryan commented:
Looks like the build is failing due to the dependency on `core_bench` for the benchmarks.

bbatsov commented on lib/core.mli:
```diff
@@ -280,6 +280,22 @@ val split : ?pos:int -> ?len:int -> re -> string -> string list
         - : string list = ["3"; "4. Commas go brrr."]
     ]}
+
+    Be careful when using [split] with zero-length patterns like [eol], [bow], 
```
Perhaps you should explain what to be careful about, as it's kind of vague even with the examples. I get the general idea, but I think this could be explained in more detail.

mooreryan replied:
I added more details in the latest commit.

mooreryan commented:
Any interest in this PR?

rgrinberg commented:
Sure, I think this PR is useful. Ping me when it's ready and I'll merge.

mooreryan commented:
Thanks @rgrinberg, I rebased, and added corresponding examples to the `split_delim` function as well, since its behavior is a bit different than `split`.

rgrinberg commented:
Thanks!
